### PR TITLE
Fixing typos - Part 3

### PR DESCRIPTION
**Pull request by Fourchaux**
**Kind:** documentation

Typos found with https://github.com/codespell-project/codespell

Grand Finale!
+350 typos found/corrected. 

Pas mal pour le Gallinacé !

### CODE REVIEW

File: proofs/refiner.ml [hunk 1]
```diff
@@ -289,7 +289,7 @@ let tclIFTHENTRYELSEMUST tac1 tac2 gl =
 (* Fails if a tactic did not solve the goal *)
 let tclCOMPLETE tac = tclTHEN tac (tclFAIL_s "Proof is not complete.")
 
-(* Try the first thats solves the current goal *)
+(* Try the first that's solves the current goal *)
```
Comment (Zimmi48) [inline proofs/refiner.ml]: Should be "that solves", shouldn't it?

File: tactics/abstract.ml [hunk 2]
```diff
@@ -99,7 +99,7 @@ let cache_term_by_tactic_then ~opaque ~name_op ?(goal_type=None) tac tacK =
   (* This is important: The [Global] and [Proof Theorem] parts of the
      goal_kind are not relevant here as build_constant_by_tactic does
      use the noop terminator; but beware if some day we remove the
-     redundancy on constrant declaration. This opens up an interesting
+     redundancy on constraint declaration. This opens up an interesting
```
Comment (Zimmi48) [inline tactics/abstract.ml]: I think this is supposed to be "constant"

File: tactics/tacticals.ml [hunk 3]
```diff
@@ -481,7 +481,7 @@ module New = struct
       ) <*>
         tclUNIT res
 
-  (* Try the first thats solves the current goal *)
+  (* Try the first that's solves the current goal *)
```
Comment (Zimmi48) [inline tactics/tacticals.ml]: Same as above: "that solves"

**Review by Zimmi48 — COMMENTED** [review]
Thanks! Just a few things to fix.

File: proofs/refiner.ml (hunk 1 above)
Comment (Fourchaux) [reply proofs/refiner.ml]: Yep. Fixed.

File: tactics/tacticals.ml (hunk 3 above)
Comment (Fourchaux) [reply tactics/tacticals.ml]: Fixed.

File: tactics/abstract.ml (hunk 2 above)
Comment (Fourchaux) [reply tactics/abstract.ml]: Allright. Fixed.

**Review by Zimmi48 — APPROVED** [review]

**Pull request merged**